solvebio/query.py

```diff
@@ -627,13 +626,7 @@ def next(self):
         self._cursor += 1
         self._buffer_idx += 1
 
-        try:
-            return self._buffer[self._buffer_idx - 1]
-        except IndexError:
-            # If this is a join query, there might not be any more
-            # items in the buffer, and no more data to fetch.
-            # TODO: Should we confirm it is a join?
-            raise StopIteration()
+        return self._buffer[self._buffer_idx - 1]
```

Comment:
Won't you still get an exception here?

Reply:
No exceptions, I added 2 tests in order to cover empty query_a/query_b. Seems it works well.